types/parameterized_decimal_type.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type ParameterizedDecimalType struct {
+	Nullability      Nullability
+	TypeVariationRef uint32
+	Precision        IntegerParam
+	Scale            IntegerParam
+}
+
+func (*ParameterizedDecimalType) isRootRef() {}
+func (m *ParameterizedDecimalType) WithNullability(n Nullability) Type {
+	m.Nullability = n
+	return m
+}
+
+func (m *ParameterizedDecimalType) GetType() Type               { return m }
+func (m *ParameterizedDecimalType) GetNullability() Nullability { return m.Nullability }
+func (m *ParameterizedDecimalType) GetTypeVariationReference() uint32 {
+	return m.TypeVariationRef
+}
+func (m *ParameterizedDecimalType) Equals(rhs Type) bool {
+	if o, ok := rhs.(*ParameterizedDecimalType); ok {
+		return *o == *m
+	}
+	return false
+}
+
+func (*ParameterizedDecimalType) ToProtoFuncArg() *proto.FunctionArgument {
+	// parameterized type are never on wire so to proto is not supported
+	panic("not supported")
+}
+
+func (m *ParameterizedDecimalType) ShortString() string {
+	t := &DecimalType{}
+	return t.ShortString()
```

Comment:
this doesn't need to be a pointer. Could probably just be 

```go
var t DecimalType
return t.ShortString()
```

Reply:
fixed it